diff_cover/violationsreporters/java_violations_reporter.py

```diff
@@ -0,0 +1,224 @@
+            for xml_document in self._xml_roots:
+                line_nodes = self._get_src_path_line_nodes(xml_document,
+                                                           src_path)
+
+                if line_nodes is None:
+                    continue
+
+                # First case, need to define violations initially
+                if violations is None:
+                    violations = set(
+                        Violation(int(line.get('num')), None)
+                        for line in line_nodes
+                        if int(line.get('count', 0)) == 0)
+
+                # If we already have a violations set,
+                # take the intersection of the new
+                # violations set and its old self
+                else:
+                    violations = violations & set(
+                        Violation(int(line.get('num')), None)
+                        for line in line_nodes
+                        if int(line.get('count', 0)) == 0
+                    )
+
+                # Measured is the union of itself and the new measured
+                measured = measured | set(
+                    int(line.get('num')) for line in line_nodes
+                )
+
+            # If we don't have any information about the source file,
+            # don't report any violations
+            if violations is None:
+                violations = set()
+            self._info_cache[src_path] = (violations, measured)
+
+    def violations(self, src_path):
+        """
+        See base class comments.
+        """
+
+        self._cache_file(src_path)
+
+        # Yield all lines not covered
+        return self._info_cache[src_path][0]
+
+    def measured_lines(self, src_path):
+        """
+        See base class docstring.
+        """
+        self._cache_file(src_path)
+        return self._info_cache[src_path][1]
+
+
+"""
+    Report checkstyle violations.
+
+    http://checkstyle.sourceforge.net/apidocs/com/puppycrawl/tools/checkstyle/DefaultLogger.html
+    https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/AuditEventDefaultFormatter.java
+"""
+checkstyle_driver = RegexBasedDriver(
+    name='checkstyle',
+    supported_extensions=['java'],
+    command=['checkstyle'],
+    expression=r'^\[\w+\]\s+([^:]+):(\d+):(?:\d+:)? (.*)$',
+    command_to_check_install=['java', '-cp', 'checkstyle-8.5-all.jar', 'com.puppycrawl.tools.checkstyle.Main', '-version']
+)
+
+
+class CheckstyleXmlDriver(QualityDriver):
+    def __init__(self):
+        """
+        See super for args
+        """
+        super(CheckstyleXmlDriver, self).__init__(
+            'checkstyle',
+            ['java'],
+            ['java', '-jar', 'checkstyle-8.5-all.jar', '-c',
+             '/google_checks.xml']
+        )
+        self.command_to_check_install = ['java', '-cp', 'checkstyle-8.5-all.jar', 'com.puppycrawl.tools.checkstyle.Main', '-version']
```

Comment:
Is there any way to have this less dependent on this specific version of checkstyle?